[Seonlo99] iP

Seonlo99 opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

### DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

### All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :stuck_out_tongue_winking_eye:

And it is **FREE**!

### Features:
 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

## If you Java programmer, you can use it to practice Java too. Here's the main method:
```java

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Seonlo99 committed:
Rename Chatbot

Seonlo99 committed:
Merge branch 'master' of https://github.com/Seonlo99/ip

Seonlo99 committed:
Add ability to store and display

Seonlo99 committed:
Add mark and  unmark task

Seonlo99 committed:
Add Todo, Deadline and Event

Seonlo99 committed:
Add automated Text UI Testing

Seonlo99 committed:
Add Handle Errors

Seonlo99 committed:
Add deletefeature

Seonlo99 committed:
Fix bug

Seonlo99 committed:
Update syntax

Seonlo99 committed:
Add Level-7 Save feature

Seonlo99 committed:
merge branch-Level-7 to master

Seonlo99 committed:
Add Level-8 Date and Time feature

Seonlo99 committed:
Add A-MoreOOP refactor related code as classes

Seonlo99 committed:
Divide classes into packages

Seonlo99 committed:
Devide classes into packages

Seonlo99 committed:
Add Gradle support for Nobita

Seonlo99 committed:
Add JUnit tests

Seonlo99 committed:
Add JavaDocs

Seonlo99 committed:
Add Level-9 find task feature

seewhyjay commented on src/main/java/nobita/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+package nobita.command;
```
I think it's really great that you implemented the stretch goal of splitting each of your commands into another class.

seewhyjay commented on src/main/java/nobita/task/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+        this.by = by;
+        String[] dueDateTime = by.split(" ");
+        this.date = LocalDate.parse(dueDateTime[0]); // input format "2016-06-11"
+        this.time = LocalTime.parse(dueDateTime[1]); // input format "06:30"
```
I think instead of having a `LocalDate` and a `LocalTime` you can just use `LocalDateTime` and have a single formatter `DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")`

seewhyjay commented on src/main/java/nobita/task/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + date.format(DateTimeFormatter.ofPattern("dd MMM yyyy")) + ")";
```
I think it might be useful to show the time as well

seewhyjay commented on src/test/java/nobita/command/MarkTaskTest.java:
```diff
@@ -0,0 +1,75 @@
+public class MarkTaskTest {
+
+    @Test
+    public void outOfRangeTest() {
```
Great negative testing

seewhyjay commented on src/test/java/nobita/parser/ParserTest.java:
```diff
@@ -0,0 +1,42 @@
+
+public class ParserTest {
+    @Test
+    public void nonIntegerTest() {
```
I think that while it is not necessary, it might be good to indicate that an exception is supposed to be thrown. This could be `markCommand_invalidIndex_exceptionThrown` like that listed [here](https://se-education.org/guides/conventions/java/intermediate.html#naming)

Seonlo99 committed:
Reformat JavaDoc

seewhyjay reviewed (commented):
Really great PR implementing many of the stretch goals.

Seonlo99 committed:
Add GUI for Nobita

VN-Hao commented on src/main/java/nobita/parser/Parser.java:
```diff
@@ -0,0 +1,87 @@
+        String[] commands = fullCommand.split(" ", 2);
+        String commandType = commands[0].toLowerCase();
+        switch (commandType) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+
+            return new MarkCommand(checkNumber(commands[1]));
+        case "unmark":
+            return new UnmarkCommand(checkNumber(commands[1]));
+        case "todo":
+            if (commands.length < 2) {
+                throw new NobitaException("The description of a todo cannot be empty.\n"
+                        + "Please specify a description.");
+            }
+            return new AddCommand(commands[1]);
+        case "deadline":
+            if (commands.length < 2) {
+                throw new NobitaException("The description of a deadline cannot be empty.\n"
+                        + "Please specify a description.");
+            }
+            String[] deadlineFields = commands[1].split(" /by ");
+            return new AddCommand(deadlineFields[0], deadlineFields[1]);
+        case "event":
+            if (commands.length < 2) {
+                throw new NobitaException("The description of a event cannot be empty.\n"
+                        + "Please specify a description.");
+            }
+            String[] eventFields = commands[1].split(" /from ");
+            String[] fromAndTo = eventFields[1].split(" /to ");
+            return new AddCommand(eventFields[0], fromAndTo[0], fromAndTo[1]);
+        case "delete":
+            return new DeleteCommand(Integer.parseInt(commands[1]) - 1);
+        case "find":
+            return new FindCommand(commands[1]);
+        default:
+            throw new NobitaException("I'm sorry, but I don't know what that means :-(");
+        }
```
Great use of **_switch_** statement. However, you can also use **_if_** statement in this case (they give the same performance though 😸) .

VN-Hao commented on src/main/java/nobita/parser/Parser.java:
```diff
@@ -0,0 +1,87 @@
+        case "list":
+            return new ListCommand();
+        case "mark":
+
```
You can remove this line!

VN-Hao commented on src/main/java/nobita/task/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Getter for number of tasks.
+     * @return An integer representing number of total tasks.
+     */
```
You should add an **empty line** between the method description and the parameter descriptions.

VN-Hao commented on src/main/java/nobita/task/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    /**
+     * Mark a task to be complete
```
You should add a **punctuation** at the end of this description here.

VN-Hao commented on src/main/java/nobita/task/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+    public TaskList findTask(String query) {
+        TaskList matchedQuery = new TaskList();
+        for (Task task: tasks) {
+            if (task.getTaskName().toLowerCase().contains(query.toLowerCase())) {
```
I think you can create a temporary boolean variable (e.g. `isMatching`) to store the boolean value of this expression to make your idea clearer!

VN-Hao reviewed (commented):
LGTM. Just a few nits to fix. 🥳

Seonlo99 committed:
Add Assert

$DeleteCommand, MarkCommand and UnmarkCommand uses a index to specify a task in tasklist.\n\nThe 3 classes assume the index to be within tasklist index.\n\nUsing Assert ensure the index is indeed within the tasklist index.

Seonlo99 committed:
Add Assert

DeleteCommand, MarkCommand and UnmarkCommand uses a index to specify a task in tasklist.\n\nThe 3 classes assume the index to be within tasklist index.\n\nUsing Assert ensure the index is indeed within the tasklist index.

Seonlo99 committed:
Add Assert

$DeleteCommand, MarkCommand and UnmarkCommand uses a index to specify a task in tasklist.\n\nThe 3 classes assume the index to be within tasklist index.\n\nUsing Assert ensure the index is indeed within the tasklist index.

Seonlo99 committed:
Add Assert

$DeleteCommand, MarkCommand and UnmarkCommand uses a index to specify a task in tasklist.\n\nThe 3 classes assume the index to be within tasklist index.\n\nUsing Assert ensure the index is indeed within the tasklist index.

Seonlo99 committed:
Add Assert

DeleteCommand, MarkCommand and UnmarkCommand uses a index to specify a task in tasklist.

The 3 classes assume the index to be within tasklist index.

Using Assert ensure the index is indeed within the tasklist index.

Seonlo99 committed:
Improve code quality

Seonlo99 committed:
Merge branch 'master' of https://github.com/Seonlo99/ip

Seonlo99 committed:
Add update feature

Seonlo99 committed:
Resolve conflict

Seonlo99 committed:
Merge pull request #1 from Seonlo99/branch-A-Assertions

Branch a assertions

Seonlo99 committed:
Add UserGuide

P

Seonlo99 committed:
Remove deadcode

Seonlo99 committed:
Add more exception handling

Seonlo99 committed:
Improve User Experience

Seonlo99 committed:
Update README.md

Seonlo99 committed:
Update README.md

Seonlo99 committed:
Update README.md

Seonlo99 committed:
Add more exception handling

Seonlo99 committed:
Merge branch 'master' of https://github.com/Seonlo99/ip
